docs/pages/blog/making-customizable-components.md

```diff
@@ -0,0 +1,289 @@
+It is a provider that’s responsible for managing all data fetching and passing that data back to components.
+
+The second important component is `ListGuesser` which defines how the data should be displayed.
+
+```jsx
+import React from 'react';
+import { Admin, Resource, ListGuesser } from 'react-admin';
+import simpleRestProvider from 'ra-data-simple-rest';
+
+const dataProvider = simpleRestProvider('https://domain.tld/api');
+
+const App = () => (
+  <Admin dataProvider={dataProvider}>
+    <Resource name="users" list={ListGuesser} />
+  </Admin>
+);
+export default App;
+```
+
+If you’re unhappy with the rendering of the `ListGuesser`, then you can define your own components by reusing smaller components.
+If you’re unhappy with the smaller components, you can replace them with custom ones, and so on.
+
+So you start with a component that’s fully functional right out of the box, and you can rewrite any of its constituent elements as needed.
+
+This approach has **one major advantage**: it gives you a lot of flexibility.
+For example, you can easily modify the order of the components and their parent/children relationships.
+
+This approach has also **one major drawback**:it gives you a lot of flexibility.
+For example, you can easily modify the order of components in a bad way.
+_The more freedom, the more bugs_.
+
+##### Drawback example
+
+To show you how easy it is to make a mistake using this technique, here is a personal example involving Material UI components.
+
+I recently tried to wrap a `TextField` component in a `FormControl`, and was frustrated when it didn’t work.
+But the reason why is quite simple: the `TextField` component is itself composed of an input wrapped inside of a `FormControl`, and neither TypeScript nor `console.error` messages could warn me that my rendered markup was redundant and broken.
+
+```jsx
+<FormControl>
+  <TextField>
+</FormControl>
+
+// Equivalent to
+
+<FormControl>
+  <FormControl>
+    <Label />
+    <Input />
+    <HelperText />
+  </FormControl>
+</FormControl>
+```
+
+This trade-off makes sense for react-admin, which is used for building complete websites.
+Their users need complete freedom when it comes to rearranging components and introducing new components anywhere.
+
+But MUI’s products exist at a lower level.
+We focus on the building blocks, not the entire website (but we do have templates for that 😉).
+So that’s not the approach we took for the `DataGrid`.
+
+#### Keep a single component
+
+On our side, we wanted to make it as simple as possible to add the DataGrid to your application, so we stuck with the individual component structure — e.g., to create a new data grid, all you need is `<DataGrid rows={...} columns={...} />`.
+
+To customize this single component, we use what we call the slot strategy.
+
+## The slot solution
+
+Now we are back to the original problem—let’s look at how we use slots to balance the freedom to customize with the need to avoid building from scratch.
```

Comment:
```suggestion
Now we are back to the original problem, how to allow deep customization on a single component. Let’s look at how we use slots to balance the freedom to customize with the need to avoid building from scratch.
```